components/script/dom/document.rs

```diff
@@ -5961,6 +5961,15 @@ fn update_with_current_instant(marker: &Cell<Option<CrossProcessInstant>>) {
     }
 }
 
+/// <https://html.spec.whatwg.org/multipage/#cors-settings-attributes>
+pub(crate) fn determine_cors_setting_for_token(token: &str) -> CorsSettings {
+    match_ignore_ascii_case! { token,
+        "anonymous" => CorsSettings::Anonymous,
+        "use-credentials" => CorsSettings::UseCredentials,
+        _ => CorsSettings::Anonymous,
+    }
+}
+
```

Comment:
It probably makes sense to make this an  implementation of `From<&str>` in `components/shared/net/request.rs` where `CorsSettings` is defined. Is there some reason why `document.rs` makes sense versus anywhere else?

Reply:
If it is specific, then please add it to an `impl` block with a name like `fn from_enumerated_attribute`.

Reply:
You mean as `CorsSettings::from_enumerated_attribute()`?